pkg/registry/core/secret/strategy.go

```diff
@@ -130,3 +133,15 @@ func SelectableFields(obj *api.Secret) fields.Set {
 	}
 	return generic.MergeFieldsSets(objectMetaFieldsSet, secretSpecificFieldsSet)
 }
+
+func warningsForSecret(secret *api.Secret) []string {
+	var warnings []string
+	if secret.Type == api.SecretTypeTLS {
+		// Verify that the key matches the cert.
+		_, err := tls.X509KeyPair(secret.Data[api.TLSCertKey], secret.Data[api.TLSPrivateKeyKey])
```

Comment:
I'm trying to think if we construct key pairs from API-provided info elsewhere, and whether passing arbitrary user-provided data into X509KeyPair exposes more of an attack surface than we have previously. @tallclair, do you happen to know?

Reply:
My feeling on this is that the permission to write secrets is high enough that we shouldn't be designing around the risk of CVEs in the go standard library code. This isn't performing a security-critical function, so the worst case would be a DoS vulnerability (which has maybe happened before with CVE-2022-24675), but I still think that given the `editor` level role required it's an acceptable risk (lots of ways to DoS a cluster with the editor role...)